### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Technyen**

### CODE REVIEW

**Comment by Technyen** [comment]
up-to-date

File: Api/.config/dotnet-tools.json
Comment (elias-rod) [inline Api/.config/dotnet-tools.json:1]: Remove this file

Comment (elias-rod) [inline Api/.config/dotnet-tools.json:1]: Remove this file

File: Api/Controllers/StoryController.cs [hunk 1]
```diff
@@ -28,19 +27,27 @@ public StoriesController(ILogger<StoriesController> logger, StoryService storySe
...
+            catch (Exception ex)
             {
-                return Conflict();
+
+                return Problem(ex.Message);
             }
+           
         }
 
         [HttpGet("getAll")]
```
Comment (elias-rod) [inline Api/Controllers/StoryController.cs]: Resolve all warnings

File: Api/Controllers/StoryController.cs [hunk 2]
```diff
@@ -71,20 +78,28 @@ public async Task<ActionResult<Story>> GetStoryAsync(string id)
...
+                else
+                {
+                    return NotFound();
+                }
             }
-            else
+            catch (Exception ex)
             {
-                return NotFound();
+
```
Comment (elias-rod) [inline Api/Controllers/StoryController.cs:101]: Remove this and all unnecessary whitespace.

File: Api/Entities/Story.cs [hunk 3]
```diff
@@ -10,5 +10,6 @@ public class Story
         public string? Category { get; set; }
         public int AgeSuggested { get; set; }
         public string? Description { get; set; }
+        public string? ImageUrl { get; set; }
```
Comment (elias-rod) [inline Api/Entities/Story.cs]: Remove null conditional operator from all models and entities that must have a value, the ones that can have a value optionally, could use the operator.

File: Api/Program.cs [hunk 4]
```diff
@@ -59,8 +64,18 @@ public static void Main(string[] args)
             builder.Services.AddSwaggerGen();
             builder.Services.AddScoped<UserService>();
             builder.Services.AddScoped<StoryService>();
-            builder.Services.AddScoped<ICosmosService, CosmosService>();
+            builder.Services.AddScoped<IRepositoryService, CosmosService>();
+            builder.Services.AddScoped<IStorageService, StorageService>();
 
+            //BlobStorage
```
Comment (elias-rod) [inline Api/Program.cs]: Remove unnecessary comments everywhere.

File: Api/Properties/PublishProfiles/Api20230927150307 - Web Deploy.pubxml [hunk 5]
```diff
@@ -6,25 +6,24 @@ by editing this MSBuild file. In order to learn more about this please visit htt
...
     <MSDeployPublishMethod>WMSVC</MSDeployPublishMethod>
     <EnableMSDeployBackup>true</EnableMSDeployBackup>
     <EnableMsDeployAppOffline>true</EnableMsDeployAppOffline>
-    <UserName>$testingelias23</UserName>
+    <UserName>$Api20230927150307</UserName>
     <_SavePWD>true</_SavePWD>
     <_DestinationType>AzureWebSite</_DestinationType>
-    <InstallAspNetCoreSiteExtension>false</InstallAspNetCoreSiteExtension>
   </PropertyGroup>
 </Project>
```
Comment (elias-rod) [inline Api/Properties/PublishProfiles/Api20230927150307 - Web Deploy.pubxml:29]: Add a new line at the end of all files.

File: Api/Services/BlobStorageService.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
+﻿using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace Api.Services
+{
+    public class StorageService : IStorageService
+	{
```
Comment (elias-rod) [inline Api/Services/BlobStorageService.cs]: Fix all files indentation.

File: Api/Services/StoryService.cs [hunk 7]
```diff
@@ -42,16 +47,20 @@ public async Task<List<Story>> GetStoriesAsync()
...
-                storyFound.Description = story.Description;
-                await _cosmosService.UpdateItemAsync(storyFound, storyFound.Id);
+                if (formFile != null)
+                {
+                    var oldExtension = storyFound.ImageUrl.Split("/").Last();
+                    await _blobStorageService.DeleteAsync(oldExtension, _imageContainerName);
+                    var newExtension = formFile.ContentType.Split("/").Last();
+                    story.ImageUrl = await _blobStorageService.UploadAsync(story.Id + "." + newExtension, formFile.OpenReadStream(), _imageContainerName);
+                }
+                await _cosmosService.UpdateItemAsync(story, story.Id);
```
Comment (elias-rod) [inline Api/Services/StoryService.cs:66]: Update the flow to consider in the EditStoryModel the ImageUrl.

File: ApiTest/UnitTest1.cs [hunk 8]
```diff
@@ -1,21 +1,23 @@
 using Api.Entities;
```
Comment (elias-rod) [inline ApiTest/UnitTest1.cs:1]: Rename file to reflect what part of the application is testing something like StoryServiceTest

File: ApiTest/UnitTest1.cs [hunk 9]
```diff
@@ -1,21 +1,23 @@
...
 namespace ApiTest
 {
     public class UnitTest1
     {
-        public readonly Mock<ICosmosService> _cosmosServiceMock = new();
+        public readonly Mock<IRepositoryService> _cosmosServiceMock = new();
+        public readonly Mock<IStorageService> _storageServiceMock = new();
 
         [Fact]
         public async void TestGetStories()
```
Comment (elias-rod) [inline ApiTest/UnitTest1.cs]: Rename to GetStoriesAsyncReturnStories

File: ApiTest/UnitTest1.cs [hunk 10]
```diff
@@ -25,31 +27,36 @@ public async void TestGetStories()
...
-            StoryService storyService = new(_cosmosServiceMock.Object);
+            _storageServiceMock.Setup(x => x.UploadAsync(It.IsAny<string>(), It.IsAny<Stream>(), "images"));
+            StoryService storyService = new(_cosmosServiceMock.Object, _storageServiceMock.Object);
 
             // Act
-            var result = await storyService.CreateStoryAsync(story);
+            var result = await storyService.CreateStoryAsync(story, (IFormFile)fileMock.Object);
            
 
             // Assert
```
Comment (elias-rod) [inline ApiTest/UnitTest1.cs:61]: Evaluate Usings file and remove if neccessary.

**Review by elias-rod — CHANGES_REQUESTED** [review]

File: Api/Services/CosmosService.cs [hunk 11]
```diff
@@ -56,17 +56,14 @@ public async Task<List<T>> GetItemsAsync<T>()
                     items.Add(item);
                 }
             }
-
             return items;
         }
 
-        public async Task<ItemResponse<T>> DeleteItemAsync<T>(string itemId, string partitionKey)
+        public async Task<ItemResponse<T>> DeleteItemAsync<T>(string? itemId, string partitionKey)
```
Comment (elias-rod) [inline Api/Services/CosmosService.cs]: Remove null operator

**Review by elias-rod — APPROVED** [review]

**Pull request merged**